src/Data/List/Permutation.agda

```diff
@@ -0,0 +1,134 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Permutation
+------------------------------------------------------------------------
+
+module Data.List.Permutation where
+
+open import Relation.Binary.PropositionalEquality as ≡
+  using (_≡_ ; refl ; setoid ; _≢_ ; isEquivalence)
+  renaming (trans to ≡-trans)
+open import Relation.Binary
+
+open import Data.List
+open import Data.List.Any
+
+open import Function
+
+open import Data.Empty
+open import Data.Product using (Σ ; _,_ ; proj₁ ; proj₂ ; _×_ ; ∃ ; ∃₂)
+
+open import Data.List.Properties
+
+module _ {a} {A : Set a} where
+
+  -- an inductive definition of permutation
+
+  data perm : Rel (List A) a where
+    unit  : ∀ {l} → perm l l
+    prep  : ∀ {l₁ l₂} x → perm l₁ l₂ → perm (x ∷ l₁) (x ∷ l₂)
+    swap  : ∀ {l₁ l₂} x y → perm l₁ l₂ → perm (x ∷ y ∷ l₁) (y ∷ x ∷ l₂)
+    trans : ∀ {l₁ l₂ l₃} → perm l₁ l₂ → perm l₂ l₃ → perm l₁ l₃
+
+  perm-sym : ∀ {l₁ l₂} → perm l₁ l₂ → perm l₂ l₁
+  perm-sym unit         = unit
+  perm-sym (prep x p)   = prep x (perm-sym p)
+  perm-sym (swap x y p) = swap y x (perm-sym p)
+  perm-sym (trans p p₁) = trans (perm-sym p₁) (perm-sym p)
+
+  perm-isEquivalence : IsEquivalence perm
```

Comment:
Similarly, this should just be called `isEquivalence` and `perm-setoid` should be called `setoid`.